packages/wdio-browserstack-service/src/service.ts

```diff
@@ -0,0 +1,240 @@
+import logger from '@wdio/logger'
+import got, { GotRequestFunction } from 'got'
+import WebDriver from 'webdriver'
+
+import { getBrowserDescription, getBrowserCapabilities, isBrowserstackCapability } from './util'
+
+const log = logger('@wdio/browserstack-service')
+
+type Capabilities = WebDriver.Capabilities & WebdriverIO.MultiRemoteCapabilities;
+
+type Browser = WebdriverIO.BrowserObject & WebdriverIO.MultiRemoteBrowserObject;
+
+type Context = any;
+
+type Feature = {
+    document: {
+        feature: {
+            name: string;
+        }
+    }
+};
+
+type Pickle = {
+    name: string;
+}
+
+type MultiRemoteAction = (sessionId: string, browserName?: string) => ReturnType<GotRequestFunction> | Promise<void>;
+
+type SessionResponse = {
+    // eslint-disable-next-line camelcase
+    automation_session: {
+        // eslint-disable-next-line camelcase
+        browser_url: string
+    }
+}
+
+export default class BrowserstackService implements WebdriverIO.ServiceInstance {
+    private _sessionBaseUrl: string = 'https://api.browserstack.com/automate/sessions';
+    private _failReasons: string[] = [];
+    private _scenariosThatRan: string[] = [];
+    private _preferScenarioName: boolean;
+    private _strict: boolean;
+    private _failureStatuses: string[] = ['failed', 'ambiguous', 'undefined', 'unknown'];
+    private _caps: Capabilities;
```

Comment:
with that this gets defined in the constructor.

Reply:
👍